taca/cleanup/cleanup.py

```diff
@@ -0,0 +1,644 @@
+    :param bool only_fastq: Remove only fastq files for closed projects
+    :param bool only_analysis: Remove only analysis data for closed projects
+    :param bool dry_run: Will summarize what is going to be done without really doing it
+    """
+    try:
+        config = CONFIG['cleanup']['irma']
+        flowcell_dir_root = config['flowcell']['root']
+        flowcell_project_source = config['flowcell']['relative_project_source']
+        data_dir = config['data_dir']
+        analysis_dir = config['analysis']['root']
+        analysis_data_to_remove = config['analysis']['files_to_remove']
+    except KeyError as e:
+        logger.error("Config file is missing the key {}, make sure it have all required information".format(str(e)))
+        raise SystemExit
+    
+    # make a connection for project db #
+    pcon = statusdb.ProjectSummaryConnection()
+    assert pcon, "Could not connect to project database in StatusDB"
+
+    #compile list for project to delete
+    project_clean_list, project_processed_list = ({}, [])
+    logger.info("Building initial project list for removing data..")
+    if only_fastq:
+        logger.info("Option 'only_fastq' is given, so will not look for analysis data")
+    elif only_analysis:
+        logger.info("Option 'only_analysis' is given, so will not look for fastq data")
+     
+    if only_analysis:
+        for pid in [d for d in os.listdir(analysis_dir) if re.match(r'^P\d+$', d) and \
+                    not os.path.exists(os.path.join(analysis_dir, d, "cleaned"))]:
+            proj_abs_path = os.path.join(analysis_dir, pid)
+            proj_info = get_closed_proj_info(pcon.get_entry(pid, use_id_view=True))
+            if proj_info and proj_info['closed_days'] >= days_analysis:
+                analysis_data, analysis_size = collect_analysis_data_irma(pid, analysis_dir, analysis_data_to_remove, return_size=True)
+                proj_info['analysis_to_remove'] = analysis_data
+                proj_info['analysis_size'] = analysis_size
+                proj_info['fastq_to_remove'] = "not_selected"
+                proj_info['fastq_size'] = 0
+                project_clean_list[proj_info['name']] = proj_info
+    else:
+        for flowcell_dir in flowcell_dir_root:
+            for fc in [d for d in os.listdir(flowcell_dir) if re.match(filesystem.RUN_RE,d)]:
+                fc_abs_path = os.path.join(flowcell_dir, fc)
+                with filesystem.chdir(fc_abs_path):
+                    projects_in_fc = [d for d in os.listdir(flowcell_project_source) \
+                                      if re.match(r'^[A-Z]+[_\.]+[A-Za-z]+_\d\d_\d\d$',d) and \
+                                      not os.path.exists(os.path.join(flowcell_project_source, d, "cleaned"))]
+                    for _proj in projects_in_fc:
+                        proj = re.sub(r'_+', '.', _proj, 1)
+                        if proj in project_processed_list:
```

Comment:
a ok... in this way you have not to query multiple time statusdb for a project you already know you need to clean.... it took me a while to understand it. Can you make this explicit with a comment. Really NICE!!!